auto_nag/scripts/bisection_without_regressed_by.py

```diff
@@ -121,34 +134,83 @@ def get_bz_params(self, date):
-            "o3": "casesubstring",
-            "v3": "his bug contains a bisection range, ",
+            "o3": "substring",
+            "v3": " this bug contains a bisection range",
             # TODO: Nag in the duplicate target instead, making sure it doesn't already have regressed_by.
             "f4": "resolution",
             "o4": "notequals",
             "v4": "DUPLICATE",
-            "emaillongdesc1": "1",
-            "emailtype1": "exact",
-            "email1": "bugmon@mozilla.com",
+            "f5": "delta_ts",
+            "o5": "greaterthan",
+            "v5": self.oldest_comment_date,
+            "f6": "OP",
+            "j6": "OR",
+            "f7": "commenter",
+            "o7": "equals",
+            "v7": "bugmon@mozilla.com",
+            "f8": "longdesc",
+            "o8": "substring",
+            "v8": "mozregression",
+            "f9": "CP",
         }
 
+    @staticmethod
+    def is_mozregression_analysis(comment_text: str):
+        """Check if the comment has a regression range from mozregression."""
+        return "ozregression" in comment_text and "pushloghtml" in comment_text
+
+    @staticmethod
+    def is_bugmon_analysis(comment_text: str):
+        """Check if the comment has a regression range from bugmon."""
+        return (
+            "BugMon: Reduced build range" in comment_text
+            or "The bug appears to have been introduced in the following build range"
+            in comment_text
+        )
+
+    @staticmethod
+    def is_bugmon_cannot_reproduce(comment_text: str):
+        """Check if the comment reports that the regression can't be reproduced by bugmon."""
+        return (
+            "Unable to reproduce" in comment_text
+            and "Removing bugmon keyword as no further action possible. Please review the bug and re-add the keyword for further analysis."
+            in comment_text
+        )
+
     def comment_handler(self, bug, bug_id, bugs):
-        range_found = False
+        analysis_comment_number = None
         # We start from the last comment just in case bugmon has updated the range.
-        for comment in bug["comments"][::-1]:
-            if (
-                "BugMon: Reduced build range" not in comment["text"]
-                and "The bug appears to have been introduced in the following build range"
-                not in comment["text"]
-            ):
+        for comment in reversed(bug["comments"]):
+            if comment["creation_time"] < self.oldest_comment_date:
+                break
+
+            # Using comments that quote other comments will lead to report an
+            # inaccurate comment number.
+            if "(In reply to " in comment["text"]:
+                continue
+
+            # There is no point in checking older comments if BugMon is not able
+            # to reproduce the bug in a newer comment.
+            if self.is_bugmon_cannot_reproduce(comment["text"]):
+                break
```

Comment:
This means the bug was fixed, but it doesn't mean the regression range was wrong. Setting the regressed by field could still be useful for tracking purposes.

Reply:
Dropped in 545cba377d6ddbb792f828f5e517e0d017a2590b